### PR TITLE
tech-debt/cognito_user_pool_domain: add sweeper

### LINKED ISSUE #14102: Acceptance test sweeper: `aws_cognito_user_pool` with dependency on domain causes error
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource

* aws_cognito_user_pool

### Description

The sweeper fails with the error message

```
Error running Sweeper (aws_cognito_user_pool) in region (us-west-2): Error deleting Cognito User Pool tf-test-9207420729894523345: InvalidParameterException: User pool cannot be deleted. It has a domain configured that should be deleted first.
```

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #14102 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
tech-debt/cognito_user_pool_domain: add sweeper
```

Output from acceptance testing:

```
go test ./aws -v -timeout=10h -sweep-allow-failures -sweep=us-west-2 -sweep-run=aws_cognito_user_pool
2020/08/13 18:04:00 [DEBUG] No Cognito user pools (i.e. domains) to sweep
2020/08/13 18:04:00 [DEBUG] Sweeper (aws_cognito_user_pool) has dependency (aws_cognito_user_pool_domain), running..
2020/08/13 18:04:00 [DEBUG] Sweeper (aws_cognito_user_pool_domain) already ran in region (us-east-1)
2020/08/13 18:04:00 [DEBUG] Running Sweeper (aws_cognito_user_pool) in region (us-east-1)
2020/08/13 18:04:00 [DEBUG] No Cognito User Pools to sweep
2020/08/13 18:04:00 Sweeper Tests ran successfully:
	- aws_cognito_user_pool_domain
	- aws_cognito_user_pool
```

### CODE REVIEW

File: aws/resource_aws_cognito_user_pool_domain_test.go [hunk 1]
```diff
@@ -13,6 +14,65 @@ import (
...
 )
 
+func init() {
+	resource.AddTestSweepers("aws_cognito_user_pool_domain", &resource.Sweeper{
+		Name: "aws_cognito_user_pool_domain",
+		F:    testSweepCognitoUserPoolDomains,
+	})
+}
+
+func testSweepCognitoUserPoolDomains(region string) error {
```
Comment (anGie44) [inline aws/resource_aws_cognito_user_pool_domain_test.go:24]: i'm not hp sure if this functionality should be implemented here on it's own or added into the `user_pool` sweeper function that already exists as they overlap quite a bit

File: aws/resource_aws_cognito_user_pool_test.go [hunk 2]
```diff
@@ -35,37 +38,32 @@ func testSweepCognitoUserPools(region string) error {
...
-		if err != nil {
-			if testSweepSkipSweepError(err) {
-				log.Printf("[WARN] Skipping Cognito User Pool sweep for %s: %s", region, err)
-				return nil
-			}
-			return fmt.Errorf("Error retrieving Cognito User Pools: %s", err)
-		}
-
-		if len(output.UserPools) == 0 {
+	err = conn.ListUserPoolsPages(input, func(resp *cognitoidentityprovider.ListUserPoolsOutput, isLast bool) bool {
```
Comment (anGie44) [inline aws/resource_aws_cognito_user_pool_test.go:41]: I came across this available function...do we have a preference over one or the other?

Comment (bflad) [reply aws/resource_aws_cognito_user_pool_test.go:41]: Paginators are always preferred! 😍

File: aws/resource_aws_cognito_user_pool_domain_test.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_cognito_user_pool_domain_test.go:24]: We've tended to lean in the direction that this is implemented (splitting out the sweepers), I think mostly to:

* Keep the sweeper deletion logic near its resource logic
* Allow us in the future to disable certain bits more easily
* Allow us in the future to acceptance test just this sweeper functionality

It does certainly feel a little weird that you list through the parent resource in this case. 😅

**Pull request merged**